jwst/assign_wcs/assign_wcs.py

```diff
@@ -50,18 +57,29 @@ def load_wcs(input_model, reference_files={}):
         wcs = WCS(pipeline)
         output_model.meta.wcs = wcs
         output_model.meta.cal_step.assign_wcs = 'COMPLETE'
-        exclude_types = ['NRS_FIXEDSLIT', 'NRS_BRIGHTOBJ', 'NRS_IFU',
-                         'NRS_MSASPEC', 'NRS_LAMP', 'MIR_MRS', 'NIS_SOSS',
-                         'NRC_WFSS', 'NRC_TSGRISM', 'NIS_WFSS', 'NRS_AUTOFLAT']
+        exclude_types = ['nrc_wfss', 'nrc_tsgrism', 'nis_wfss',
+                         'nrs_fixedslit', 'nrs_ifu', 'nrs_msaspec',
+                         'nrs_autowave', 'nrs_autoflat', 'nrs_lamp',
+                         'nrs_brightobj', 'mir_lrs-fixedslit', 'mir_lrs-slitless',
+                         'mir_mrs', 'nis_soss']
 
-        if output_model.meta.exposure.type not in exclude_types:
-            orig_s_region = output_model.meta.wcsinfo.s_region.strip()
-            update_s_region(output_model)
-            if orig_s_region != output_model.meta.wcsinfo.s_region.strip():
-                log.info("assign_wcs updated S_REGION to {0}".format(output_model.meta.wcsinfo.s_region))
-        else:
-            log.info("assign_wcs did not update S_REGION for type {0}".format(output_model.meta.exposure.type))
-        log.info("COMPLETED assign_wcs")
+        if output_model.meta.exposure.type.lower() not in exclude_types:
+            if output_model.meta.exposure.type.lower() in IMAGING_TYPES:
+                try:
+                    update_s_region_imaging(output_model)
+                except:
```

Comment:
What kinda of error are you expecting here?  ValueError?  Generic Exception?  If so, it might be good to catch it:
```
except Exception as error:
    log.error("Unable to update S_REGION for type  %s: %s", output_model.meta.exposure.type, error)
```
Similar with the bare `except` blocks below.